mono/mini/method-to-ir.c

```diff
@@ -10138,7 +10149,7 @@ mono_method_to_ir (MonoCompile *cfg, MonoMethod *method, MonoBasicBlock *start_b
 
 			/* Generate IR to do the actual load/store operation */
 
-			if ((il_op == MONO_CEE_STFLD || il_op == MONO_CEE_STSFLD) && (ins_flag & MONO_INST_VOLATILE)) {
+			if (store_val && (ins_flag & MONO_INST_VOLATILE)) {
```

Comment:
either I don't understand this change or it's unrelated to what this PR is about. could you elaborate?